From e39a240189580e5a2f095c6e6f30f9d6ef35cb39 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 30 Dec 2021 15:42:20 +0800
Subject: [PATCH 19/19] qwaylandwindow: Fix a memory leak in create/destroy

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/client/qwaylandwindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 46475fd..ffe1b2c 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -103,6 +103,8 @@ QWaylandWindow::~QWaylandWindow()
     if (mMouseGrab == this) {
         mMouseGrab = nullptr;
     }
+
+    wl_event_queue_destroy(mFrameQueue);
 }
 
 void QWaylandWindow::ensureSize()
-- 
2.20.1

